churchland_pipeline_python/acquisition.py

```diff
@@ -163,19 +165,56 @@ def populate(self,
                         print('Missing notes for session {}'.format(date))
                     else:
                         with open(session_path + notes_files,'r') as f:
-                            self.Notes.insert1((date, monkey, 0, f.read()))
+                            self.Notes.insert1(dict(**session_key, session_notes_id=0, session_notes=f.read()))
 
                     # insert neural signal processor
-                    self.Hardware.insert1(dict(
-                        session_date=date,
-                        monkey=monkey,
-                        **(equipment.Hardware & {'hardware': neural_signal_processor}).fetch1('KEY')
-                        ))
+                    self.Hardware.insert1(dict(**session_key, **nsp_key))
 
 # -------------------------------------------------------------------------------------------------------------------------------
 # LEVEL 2
 # -------------------------------------------------------------------------------------------------------------------------------
 
+@schema
+class BehaviorRecording(dj.Imported):
+    definition = """
+    # Behavior recording, imported from Speedgoat files
+    -> Session
+    ---
+    behavior_summary_file_path: varchar(1012) # path to summary file (temporary)
+    behavior_sample_rate = 1e3: smallint unsigned # sampling rate for behavioral data (Hz)
+    """
+
+    key_source = Session - 'session_problem'
+    
+    def make(self, key):
+        
+        session_key = (Session & key).fetch1()
+        
+        # path to raw data
+        raw_path = Session.getrawpath(session_key['monkey'],session_key['rig'],session_key['task']) + str(key['session_date'])
+        
+        # identify task controller
+        task_controller_hardware = (Task & Session & session_key).fetch1('task_controller_hardware')
+
+        if task_controller_hardware == 'Speedgoat':
+
+            # path to speedgoat files
+            speedgoat_path = raw_path + '/speedgoat/'
+            speedgoat_files = list(os.listdir(speedgoat_path))
+
+            # speedgoat summary file
+            summary_file = next(filter(lambda f: re.search('.*\.summary',f), speedgoat_files))
```

Comment:
```suggestion
            summary_file = next(f for f in spedgoat_files if re.search('.*\.summary',f))
```
Somehow the `filter` function is rarely more expressive than a simple generator expression.